partner_communication/models/communication_job.py

```diff
@@ -745,7 +741,13 @@ def _send_mail(self):
         final_state = "pending"
         if email.state == "sent":
             if self.need_call == "after_sending":
-                final_state = "call"
+                # same as in the send method
+                self.activity_schedule('mail.mail_activity_data_call',
+                                       summary="call_" + self.need_call,
+                                       user_id=self.user_id.id,
+                                       note="Call {} at (phone) {}"
+                                       .format(partner.name,
+                                               partner.phone))
```

Comment:
This has to be removed I think, it is already handled in your code in the `send` method, regardless if the communication is sent by email or printed.